### PR TITLE
Fix Kokkos_CoreUnitTest_DeviceAndThreads

### LINKED ISSUE #6326: Kokkos_CoreUnitTest_DeviceAndThreads Failure across Diverse Machines, and Python Installations
Here is the error (`loader.py` cannot find the `TestDeviceAndThreads.py` module):

```
/usr/workspace/prox_snl/Kokkos/kokkos/build_check_hip_5699/core/unit_test/TestDeviceAndThreads (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: /usr/workspace/prox_snl/Kokkos/kokkos/build_check_hip_5699/core/unit_test/TestDeviceAndThreads (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: /usr/workspace/prox_snl/Kokkos/kokkos/build_check_hip_5699/core/unit_test/TestDeviceAndThreads
Traceback (most recent call last):
  File "/opt/cray/pe/python/3.10.10/lib/python3.10/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
ModuleNotFoundError: No module named '/usr/workspace/prox_snl/Kokkos/kokkos/build_check_hip_5699/core/unit_test/TestDeviceAndThreads'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)

```

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #6326. I am seeing
```
28: Test command: /usr/bin/python3.9 "-m" "unittest" "-v" "/home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads.py"
28: Test timeout computed to be: 1500
28: /home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads (unittest.loader._FailedTest) ... ERROR
28: 
28: ======================================================================
28: ERROR: /home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads (unittest.loader._FailedTest)
28: ----------------------------------------------------------------------
28: ImportError: Failed to import test module: /home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads
28: Traceback (most recent call last):
28:   File "/usr/lib64/python3.9/unittest/loader.py", line 154, in loadTestsFromName
28:     module = __import__(module_name)
28: ModuleNotFoundError: No module named '/home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads'
28: 
28: 
28: ----------------------------------------------------------------------
28: Ran 1 test in 0.000s
28: 
28: FAILED (errors=1)
1/1 Test #28: Kokkos_CoreUnitTest_DeviceAndThreads ...***Failed    0.03 sec
```
when trying to test with GH200. It seems that executing the test file directly works.

### CODE REVIEW

**Review by ajpowelsnl — APPROVED** [review]
Looks good -- I'm assuming this change fixes Python module import errors?

**Review by dalg24 — COMMENTED** [review]
You're going to need to do more explaining

`python -m unittest` is the recommended way to run as far as I can tell.
Also we definitely meant the verbose flag.

**Comment by dalg24** [comment]
```python
if __name__ == '__main__':
    print("hello from main")
    unittest.main()
```
prints when the `-m unittest` is omitted.
So I suppose we could drop it but I want the `-v`.

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -1277,7 +1277,7 @@ if (NOT KOKKOS_HAS_TRILINOS)
     )
     add_test(
       NAME Kokkos_CoreUnitTest_DeviceAndThreads
-      COMMAND ${Python3_EXECUTABLE} -m unittest -v $<TARGET_FILE_DIR:Kokkos_CoreUnitTest_DeviceAndThreads>/TestDeviceAndThreads.py
+      COMMAND ${Python3_EXECUTABLE} $<TARGET_FILE_DIR:Kokkos_CoreUnitTest_DeviceAndThreads>/TestDeviceAndThreads.py
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt:1280]:
```suggestion
      COMMAND ${Python3_EXECUTABLE} $<TARGET_FILE_DIR:Kokkos_CoreUnitTest_DeviceAndThreads>/TestDeviceAndThreads.py -v
```

**Comment by masterleinad** [comment]
The output with this version is
```
    Start 23: Kokkos_CoreUnitTest_DeviceAndThreads

23: Test command: /opt/homebrew/Frameworks/Python.framework/Versions/3.12/bin/python3.12 "/tmp/kokkos/build/core/unit_test/TestDeviceAndThreads.py"
23: Working Directory: /tmp/kokkos/build/core/unit_test
23: Test timeout computed to be: 1500
23: test_device_id (__main__.KokkosInitializationTestCase.test_device_id) ... skipped 'no device backend enabled'
23: test_disable_warnings (__main__.KokkosInitializationTestCase.test_disable_warnings) ... ok
23: test_num_devices (__main__.KokkosInitializationTestCase.test_num_devices) ... skipped 'no device backend enabled'
23: test_num_threads (__main__.KokkosInitializationTestCase.test_num_threads) ... ok
23: test_tune_internals (__main__.KokkosInitializationTestCase.test_tune_internals) ... ok
23: 
23: ----------------------------------------------------------------------
23: Ran 5 tests in 0.481s
23: 
23: OK (skipped=2)
2/2 Test #23: Kokkos_CoreUnitTest_DeviceAndThreads ...   Passed    0.56 sec
```
but we can add "-v" and omit the verbosity in main.

**Comment by masterleinad** [comment]
> `python -m unittest` is the recommended way to run as far as I can tell.

I don't have a good explanation for why the test failed on that particular platform/build. This alternative does the same thing and works.

**Comment by masterleinad** [comment]
> The PR is arguing you saw failures on GH200 but the output you posted seems to be from your MacBook. Did you confirm that the latest version resolves the issue you saw on GH200?

Yes
```
28: Test command: /usr/bin/python3.9 "/home/users/darndt/kokkos/build/core/unit_test/TestDeviceAndThreads.py" "-v"
28: Test timeout computed to be: 1500
28: test_device_id (__main__.KokkosInitializationTestCase) ... ok
28: test_disable_warnings (__main__.KokkosInitializationTestCase) ... ok
28: test_num_devices (__main__.KokkosInitializationTestCase) ... ok
28: test_num_threads (__main__.KokkosInitializationTestCase) ... skipped 'no host parallel backend enabled'
28: test_tune_internals (__main__.KokkosInitializationTestCase) ... ok
28: 
28: ----------------------------------------------------------------------
28: Ran 5 tests in 4.257s
28: 
28: OK (skipped=1)
2/2 Test #28: Kokkos_CoreUnitTest_DeviceAndThreads ...   Passed    4.29 sec
```

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Maybe improve your description/title and say something like "avoid running unittest library as a script in python device and threads test"

**Comment by ajpowelsnl** [comment]
> > `python -m unittest` is the recommended way to run as far as I can tell.
> 
> I don't have a good explanation for why the test failed on that particular platform/build. This alternative does the same thing and works.

FWIW, I've also seen the same failure on various V100 machines.

**Comment by masterleinad** [comment]
> FWIW, I've also seen the same failure on various V100 machines.

Does this pull request resolve the issue for you?

**Comment by ajpowelsnl** [comment]
> > FWIW, I've also seen the same failure on various V100 machines.
> 
> Does this pull request resolve the issue for you?

I will try it, and let you know.  BTW, to be a bit clearer, it was sometime in the last year or so that I've observed this failure.

**Pull request merged**